### PR TITLE
Add unsqueeze and transpose patterns

### PR DESCRIPTION
**Pull request by kurapov-peter**
Both rely on the second input to be a constant.

Depends on #177

### CODE REVIEW

File: src/common/transformations/src/transformations/mlir/op/unsqueeze.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                shape[i] = 1;
+                coord_idx++;
+            } else {
+                const auto& dim = ov_input_shape[input_idx];
+                shape[i] = dim.is_dynamic() ? ShapedType::kDynamic : dim.get_length();
+                input_idx++;
+                expand_groups.push_back(group);
+                group = ReassociationIndices();
+            }
+        }
```
Comment (slyalin) [inline src/common/transformations/src/transformations/mlir/op/unsqueeze.cpp:55]: Why cannot we just take it as `node->get_output_partial_shape(0)`? I believe the shape in MLIR should match the shape from OV.

Comment (kurapov-peter) [reply src/common/transformations/src/transformations/mlir/op/unsqueeze.cpp:55]: Would it be always available? (dynamic shapes?)

Comment (kurapov-peter) [reply src/common/transformations/src/transformations/mlir/op/unsqueeze.cpp:55]: I'd still need to generate the reassociation groups anyway though

Comment (kurapov-peter) [reply src/common/transformations/src/transformations/mlir/op/unsqueeze.cpp:55]: I'd still need to generate the reassociation groups anyway though

Comment (slyalin) [reply src/common/transformations/src/transformations/mlir/op/unsqueeze.cpp:55]: `get_output_partial_shape` always returns shape, including dynamic shapes, which means a part of the dimensions can be dynamic and you already know how to detect them -- `dim.is_dynamic()`. But I agree that you still need to build reassociation groups.

**Pull request merged**